pkg/sql/types/types.go

```diff
@@ -1923,6 +1924,19 @@ func (t *T) SQLString() string {
 	return strings.ToUpper(t.Name())
 }
 
+// SQLStringForError returns a version of SQLString that will preserve safe
+// information during redaction. It is suitable for usage in error messages.
+func (t *T) SQLStringForError() redact.RedactableString {
+	if t.UserDefined() {
+		return redact.Sprintf("UDT: %s", t.SQLString())
+	}
+	if t.Family() == EnumFamily {
+		return redact.Sprintf("enum: %s", t.SQLString())
+	}
+	// Other types are safe to leave un-redacted.
+	return redact.Sprint(redact.SafeString(t.SQLString()))
```

Comment:
Right now as written you are promising that any current _and future_ type implementations will produce safe representations. How do you plan to make good for that promise, for everyone ever adding a new type family in the future?

If i were you i would invert.this logic, and enumerate explicitly the types that are safe, and then fall back to the other case for everything else.

Reply:
Done. I also added a test to ensure expected output.